File: CodeForces/271a/main.py
def unique(n):
    mp = {}
    while (n):
        dig = n % 10
        if (dig in mp):
            return False
        else:
            mp[dig] = True
        n //= 10

    return True


def solve():
    s = int(input().strip())
    flag = True
    while (flag):

        s += 1
        if (unique(s)):
            print(s)
            flag = False

    return -1


solve()

File: CodeForces/271a/main.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

bool unique(int n) {
  unordered_map<int, bool> mp;
  while (n) {
    int dig = n % 10;
    if (mp.find(dig) != mp.end()) {
      return false;
    } else {
      mp[dig] = true;
    }
    n /= 10;
  }
  return true;
}

int main() {
  bool flag = true;
  int n;
  cin >> n;

  while (flag) {
    n += 1;
    if (unique(n)) {
      cout << n;
      flag = false;
    }
  }

  return 0;
}
